presto-spi/src/main/java/com/facebook/presto/spi/type/Type.java

```diff
@@ -146,6 +146,11 @@
     boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition);
 
     /**
+     * Are the values in the specified blocks at the specified positions equal are not distinct?
```

Comment:
can you explain a bit more in detail how isNotDistinctFrom is different than equalTo? from what I remember, equalTo at the type level actually is just isNotDistinctFrom (hence the null checks when using equalTo).